Fix namedtensor ci

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26060 Turn on BUILD_NAMEDTENSOR permanently
* **#26257 Fix namedtensor ci**

In native_functions.yaml, all overloads must have unique overload names.
This PR fixes `flatten` to have unique names for the overloads.

Test Plan:
- tested locally, but also [namedtensor ci]

Differential Revision: [D17391243](https://our.internmc.facebook.com/intern/diff/D17391243)

zou3519 committed:
Fix namedtensor ci

In native_functions.yaml, all overloads must have unique overload names.
This PR fixes `flatten` to have unique names for the overloads.

Test Plan:
- tested locally, but also [namedtensor ci]

[ghstack-poisoned]

smessmer reviewed (approved):
looks good, please make overload names more meaningful

zou3519 committed:
Update on "Fix namedtensor ci"


In native_functions.yaml, all overloads must have unique overload names.
This PR fixes `flatten` to have unique names for the overloads.

Test Plan:
- tested locally, but also [namedtensor ci]

Differential Revision: [D17391243](https://our.internmc.facebook.com/intern/diff/D17391243)

[ghstack-poisoned]

zou3519 committed:
Update on "Fix namedtensor ci"


In native_functions.yaml, all overloads must have unique overload names.
This PR fixes `flatten` to have unique names for the overloads.

Test Plan:
- tested locally, but also [namedtensor ci]

Differential Revision: [D17391243](https://our.internmc.facebook.com/intern/diff/D17391243)

[ghstack-poisoned]
